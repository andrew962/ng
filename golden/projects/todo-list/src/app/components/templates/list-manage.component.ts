import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { MatDialog } from '@angular/material/dialog';
import { MatSelectionListChange } from '@angular/material/list';
import { NewItemList } from '../../models/newitem.model';
import { DialogCreateNewComponent } from './dialog-create-new.component';

@Component({
  selector: 'app-list-manage',
  templateUrl: './list-manage.component.html',
  styles: [
  ]
})
export class ListManageComponent implements OnInit {
  @Input() listItems: NewItemList[];
  @Output() listItemsChange = new EventEmitter<NewItemList[]>();

  @Output() itemSelectedChange = new EventEmitter();
  constructor(
    private dialog: MatDialog,) { }

  ngOnInit(): void {
  }

  updateList(list) {
    this.listItems = list;
    this.listItemsChange.emit(list);
  }

  itemSelected(event) {
    this.itemSelectedChange.emit(event);
  }
  btnAddNote() {
    let dialogRef = this.dialog.open(DialogCreateNewComponent);
    dialogRef.afterClosed().subscribe((res: NewItemList) => {
      if (res.Title != null) {
        this.listItems.push(res);
      }
    });
  }
  btnDelete(i: number) {
    this.listItems.splice(i, 1);
  }
}
