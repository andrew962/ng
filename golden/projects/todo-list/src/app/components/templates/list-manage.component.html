<!-- <mat-list>
    <mat-list-item *ngFor="let item of listItems; let i=index">
        <mat-card>
            <div class="col-6 p-0">
                <div mat-line>{{ item.Title}}</div>
                <div mat-line> {{item.Content}} </div>
            </div>
            <div class="col-6 text-right p-0">
                <div class="btn-group" role="group" aria-label="Button group">
                    <button class="btn btn-primary btn-sm" (click)="itemSelected(item)"><i
                            class="fas fa-edit"></i></button>
                    <button class="btn btn-danger btn-sm" (click)="btnDelete(i)"><i class="fas fa-trash"></i></button>
                </div>
            </div>
        </mat-card>
    </mat-list-item>
</mat-list> -->
<ng-container *ngIf="listItems.length<=0">
    <h3 class="d-block">
        Nota nueva
    </h3>
    <input type="image" class="shadow p-2" (click)="btnAddNote()" width="50%" src="assets/img/add_notes.svg"
        alt="Lista de notas">
</ng-container>